core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -267,6 +268,25 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
       emitMetrics()
   }
 
+  /** Resend next item in the buffer, or trigger next item in the feed, if no items in the buffer. */
+  def processBufferOrFeed() = {
+    // If buffer has more items, and head has not already been resent, send next one, otherwise get next from feed.
+    runBuffer.dequeueOption match {
+      case Some((run, _)) => //run the first from buffer
+        implicit val tid = run.msg.transid
+        //avoid sending dupes
+        if (resent.isEmpty) {
+          logging.info(this, s"re-processing from buffer (${runBuffer.length} items in buffer)")
+          resent = Some(run)
+          self ! run
+        } else {
+          //do not resend the buffer head multiple times (may reach this point from multiple messages, before the buffer head is re-processed)
```

Comment:
here a message is dequeued from the buffer and seemingly dropped - shouldn't this message get put back in the the run buffer?

Reply:
Yeah this is confusing - dequeueOption does not alter runBuffer. So if a message is resent, it also remains at the head of buffer until https://github.com/apache/openwhisk/pull/4593/files/f84fa581a910d1d6042a26e5244674251e056fe5#diff-726b36b3ab8c7cff0b93dead84311839R181

This part (head of buffer is stuck there till it can be processed) has not changed, just way in which retries (`self ! run`) are attempted is much fewer. The hyperactive retries were likely also the reason for the "only log the rescheduling message once per 10s" logic, but I tried to leave that log handling in place for now.